src/state/toasts.ts

```diff
@@ -26,109 +27,236 @@ type ToastData = ToastParams & {
+        async getItem(name: string) {
+          const storeToasts = window.localStorage.getItem(name)
+          const storeAccount = window.localStorage.getItem("account")
+
+          if (storeAccount == null) return storeToasts
+
+          const { state: account } = JSON.parse(storeAccount)
+
+          const accountAddress = account?.account.address
+
+          if (accountAddress) {
+            const accountToastsDeprecated = window.localStorage.getItem(
+              `toasts_${accountAddress}`,
+            )
+
+            const toastsDeprecated =
+              accountToastsDeprecated &&
+              JSON.parse(accountToastsDeprecated).map((toast: ToastData) => ({
+                ...toast,
+                hidden: true,
+              }))
+
+            if (storeToasts != null) {
+              const { state: toastsState } = JSON.parse(storeToasts)
+
+              const allToasts = { ...toastsState.toasts }
+
+              const accountToasts = allToasts[accountAddress]
+
+              if (!accountToasts) {
+                if (toastsDeprecated != null) {
+                  allToasts[accountAddress] = toastsDeprecated
+
+                  window.localStorage.removeItem(`toasts_${accountAddress}`) // remove deprecated storage
+                } else {
+                  allToasts[accountAddress] = []
+                }
+              }
+              return JSON.stringify({
+                ...toastsState,
+                state: { toasts: allToasts },
+              })
+            } else {
+              if (toastsDeprecated != null) {
+                window.localStorage.removeItem(`toasts_${accountAddress}`) // remove deprecated storage
+                return JSON.stringify({
+                  version: 0,
+                  state: {
+                    toasts: {
+                      [accountAddress]: toastsDeprecated,
+                    },
+                  },
+                })
+              } else {
+                return JSON.stringify({
+                  version: 0,
+                  state: {
+                    toasts: {
+                      [accountAddress]: [],
+                    },
+                  },
+                })
+              }
+            }
+          }
+          return storeToasts
+        },
+        setItem(name, value) {
+          const parsedState = JSON.parse(value)
+
+          const stringState = JSON.stringify({
+            version: parsedState.version,
+            state: { toasts: parsedState.state.toasts },
+          })
+
+          window.localStorage.setItem(name, stringState)
+        },
+        removeItem(name) {
+          window.localStorage.removeItem(name)
+        },
+      })),
+    },
+  ),
+)
+
+export const useToast = () => {
+  const store = useToastsStore()
+  const { account } = useAccountStore()
+
+  const toasts = useMemo(() => {
+    if (account?.address) {
+      const toasts = store.toasts[account?.address]
```

Comment:
```suggestion
      const toasts = store.toasts[account.address]
```

should be defined if it's nested in `if(account?.address)` condition.